tool/tsh/tsh.go

```diff
@@ -720,6 +720,7 @@ func Run(ctx context.Context, args []string, opts ...cliOption) error {
 	lsApps.Flag("all", "List apps from all clusters and proxies.").Short('R').BoolVar(&cf.ListAll)
 	appLogin := apps.Command("login", "Retrieve short-lived certificate for an app.")
 	appLogin.Arg("app", "App name to retrieve credentials for. Can be obtained from `tsh apps ls` output.").Required().StringVar(&cf.AppName)
+	appLogin.Flag("cluster", clusterHelp).Short('c').StringVar(&cf.SiteName)
```

Comment:
Good call, will do.

Reply:
Note that this only makes sense after `https://github.com/gravitational/teleport/pull/26239` is merged. Right now we need a cert for the app, and we can only get it from `tsh app login` which already allows you to specify the cluster.

Reply:
In case of TCP apps (or when local proxy is required), `tsh app login` will print out to use `tsh proxy app`. So `tsh app login --cluster leaf <tcp-app>` will print something not usable. Same applies to aws/azure/gcp.